ceph_deploy/hosts/remotes.py

```diff
@@ -200,9 +200,15 @@ def write_monitor_keyring(keyring, monitor_keyring):
     write_file(keyring, monitor_keyring)
 
 
-def write_file(path, content):
+def write_file(path, content, mode=None, directory=None):
+    if directory:
+        if path.startswith("/"):
+            path = path[1:]
+        path = os.path.join(directory, path)
     with file(path, 'w') as f:
         f.write(content)
+    if mode:
+        os.chmod(path, mode)
```

Comment:
Thanks Sage.  This thought crossed my mind, and I even looked for the right Python call to do this (open() and file() built-ins don't take a file permissions mode).

I'll have to drop to a lower-level os.open() to do it, I think.  :)